wimm.Secundatives/Variant.cs

```diff
@@ -0,0 +1,82 @@
+﻿using System;
+
+namespace wimm.Secundatives
+{
+    /// <summary>
+    /// A typesafe union that contains either a <typeparamref name="T"/> or a <typeparamref name="U"/>
+    /// </summary>
+    public class Variant<T, U>
+    {
+        /// <summary>
+        /// Ensures that <typeparamref name="U"/> and <typeparamref name="T"/> are not within the same
+        /// inheritance heirarchy
+        /// </summary>
+        static Variant()
+        {
+            var uType = typeof(U);
+            var tType = typeof(T);
+
+            if (tType.IsSubclassOf(uType) || uType.IsSubclassOf(tType) )
```

Comment:
does this only not because of the casting?

Reply:
Yeah. Variant<Error,SubError> would always return true for Is<Error>

Reply:
Is<Error>**

Reply:
could use runtime type instead of is? wouldn't break the interface to go from disallowing to allowing so wont block either way though